charts/redpanda/templates/tests/test-prometheus-targets.yaml

```diff
@@ -0,0 +1,50 @@
+{{/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+  
+  http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  */}}
+{{- if .Values.monitoring.enabled }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ include "redpanda.fullname" . }}-test-prometheus-targets"
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+  {{- with include "full.labels" . }}
+  {{- . | nindent 4 }}
+  {{- end }}
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-delete-policy": before-hook-creation
+spec:
+  restartPolicy: Never
+  securityContext:
+    runAsUser: 65535
+    runAsGroup: 65535
+  {{- with .Values.imagePullSecrets }}
+  imagePullSecrets: {{- toYaml . | nindent 4 }}
+  {{- end }}
+  containers:
+    - name: {{ template "redpanda.name" . }}
+      image: mintel/docker-alpine-bash-curl-jq:latest
+      command: [ "/bin/bash", "-c" ]
+      args:
+        - |
+          set -xe
+          RESULT=$( curl --fail http://prometheus-operated.default.svc.cluster.local:9090/api/v1/targets?status=active | jq . | grep '"health": "up"' | wc -l  )
+          if [ $RESULT != {{ .Values.statefulset.replicas }} ]; then 
+              echo "targets are down, only have:\n ${RESULT} targets 'up'"
+              exit 1
+          fi
```

Comment:
From the test output
```
  + RESULT=21
  + '[' 21 '!=' 3 ']'
  + echo 'targets are down, only have:\n 21 targets '\''up'\'''
  + exit 1
  targets are down, only have:\n 21 targets 'up'
```
I wonder if targets have namespace label, so that you can compare correct targets. 21 number seems odd to me.

Reply:
been trying to get this right. ill check again. The prometheus api does not make this easy, so parsing is required

Reply:
this is working locally, strangely enough, given the same setup...